adabot/circuitpython_libraries.py

```diff
@@ -842,110 +861,140 @@ def print_issue_overview(*insights):
-        "issue_authors": set(),
-        "issue_closers": set(),
-    }
-    core_insights = copy.deepcopy(lib_insights)
-    for k in core_insights:
-        if isinstance(core_insights[k], set):
-            core_insights[k] = set()
-        if isinstance(core_insights[k], list):
-            core_insights[k] = []
-    repo_needs_work = []
-    since = datetime.datetime.now() - datetime.timedelta(days=7)
-    repos_by_error = {}
-
-    for repo in repos:
-        errors = validate_repo(repo)
-        if errors:
-            need_work += 1
-            repo_needs_work.append(repo)
-            # print(repo["full_name"])
-            # print("\n".join(errors))
-            # print()
-        for error in errors:
-            if error not in repos_by_error:
-                repos_by_error[error] = []
-            repos_by_error[error].append(repo["html_url"])
-        insights = lib_insights
-        if repo["name"] == "circuitpython" and repo["owner"]["login"] == "adafruit":
-            insights = core_insights
-        gather_insights(repo, insights, since)
-    print()
-    print("State of CircuitPython + Libraries")
-
-    print("Overall")
-    print_pr_overview(lib_insights, core_insights)
-    print_issue_overview(lib_insights, core_insights)
-
-    print()
-    print("Core")
-    print_pr_overview(core_insights)
-    print("* {} open pull requests".format(len(core_insights["open_prs"])))
-    for pr in core_insights["open_prs"]:
-        print("  * {}".format(pr))
-    print_issue_overview(core_insights)
-    print("* {} open issues".format(len(insights["open_issues"])))
-    print("  * https://github.com/adafruit/circuitpython/issues")
-    print()
-    print_circuitpython_download_stats()
-
-    print()
-    print("Libraries")
-    print_pr_overview(lib_insights)
-    print("* {} open pull requests".format(len(lib_insights["open_prs"])))
-    for pr in lib_insights["open_prs"]:
-        print("  * {}".format(pr))
-    print_issue_overview(lib_insights)
-    print("* {} open issues".format(len(lib_insights["open_issues"])))
-    for issue in lib_insights["open_issues"]:
-        print("  * {}".format(issue))
-
-    lib_repos = []
-    for repo in repos:
-        if repo["owner"]["login"] == "adafruit" and repo["name"].startswith("Adafruit_CircuitPython"):
-            lib_repos.append(repo)
-
-    # print("- [ ] [{0}](https://github.com/{1})".format(repo["name"], repo["full_name"]))
-    print("{} out of {} repos need work.".format(need_work, len(lib_repos)))
-
-    list_repos_for_errors = [ERROR_NOT_IN_BUNDLE]
-
-    for error in repos_by_error:
-        if not repos_by_error[error]:
-            continue
-        print()
-        error_count = len(repos_by_error[error])
-        print("{} - {}".format(error, error_count))
-        if error_count <= 5 or error in list_repos_for_errors:
-            print("\n".join(["* " + x for x in repos_by_error[error]]))
+    output_handler("Latest pylint is: {}".format(latest_pylint))
+
+    try:
```

Comment:
Whats this for? I think the internal block should be a separate function to make the try catch easier to read. Thanks!

Reply:
so, just move all the original code to something like `def run_library_checks():`? easy enough..